extensions/ql-vscode/src/vscode-tests/test-dispose-handler.ts

```diff
@@ -0,0 +1,14 @@
+import { Disposable } from 'vscode';
+import { DisposableObject } from '../pure/disposable-object';
+
+export function testDisposeHandler(disposable: any & Disposable) {
+  if (disposable.onDidExpandElement && disposable.onDidCollapseElement && disposable.reveal) {
+    // This looks like a treeViewer. Don't dispose
+    return;
+  }
+  if (disposable instanceof DisposableObject) {
+    disposable.dispose(testDisposeHandler);
```

Comment:
Doesn't this loop between the call to `disposable.dispose(testDisposeHandler)` here and to `disposeHandler(trackedObject)` in `DisposableObject.dispose`?

Reply:
No. If a dispose handler is used, the `dispose` method of the `DisposableObject` never calls `dispose` directly on its children. Rather, it gets delegated to the handler.

Reply:
Oh yes a `DisposableObject` has disposable _children_. Ok cool. Your tests will expose it if we break this.